controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -90,11 +90,26 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	case len(instances.Items) == 1:
 		instance = &instances.Items[0]
 	case len(instances.Items) > 1:
-		message := fmt.Sprintf("only one instance of DSCInitialization object is allowed. Update existing instance name %s", req.Name)
-		_, _ = r.updateStatus(ctx, &instances.Items[0], func(saved *dsciv1.DSCInitialization) {
-			status.SetErrorCondition(&saved.Status.Conditions, status.DuplicateDSCInitialization, message)
-			saved.Status.Phase = status.PhaseError
-		})
+		// find out the one by created timestamp and use it as the default one
+		markedWrongDSCI := []dsciv1.DSCInitialization{}
+		earliestDSCI := &instances.Items[0]
+		for _, instance := range instances.Items {
+			currentDSCI := instance
+			if currentDSCI.CreationTimestamp.Before(&earliestDSCI.CreationTimestamp) || currentDSCI.CreationTimestamp.Equal(&earliestDSCI.CreationTimestamp) {
+				earliestDSCI = &currentDSCI
+			} else {
+				markedWrongDSCI = append(markedWrongDSCI, currentDSCI)
+			}
+		}
+		message := fmt.Sprintf("only one instance of DSCInitialization object is allowed. Please delete other instances than %s", earliestDSCI.Name)
+		// update all instances Message and Status
+		for _, deletionInstance := range markedWrongDSCI {
+			_, _ = r.updateStatus(ctx, &deletionInstance, func(saved *dsciv1.DSCInitialization) {
+				status.SetErrorCondition(&saved.Status.Conditions, status.DuplicateDSCInitialization, message)
+				saved.Status.Phase = status.PhaseError
+			})
+		}
```

Comment:
Assuming we have two DSCIs [`new`, `old`] in `instances` slice, here's what will happen: 

- `earliestDSCI` will initially be `new`
- in the loop:
  - first iteration: `currentDSCI` is `new`, which is also `earliestDSCI` so it will pass `if` clause and stay the same
  - second iteration:  `currentDSCI` is `old` which is before `new` so it will now become new `earliestDSCI` (enters `if` branch, not `else`)

And we are left with empty `markedWrongDSCI` so nothing will be reported :(

Generally speaking, I think it's time to introduce a webhook and implement  `ValidateCreate` from `webhook.Validator` interface. We should not even reach reconcile in such a case nor think in the terms of multiple instances of DSCIs. Validation webhook will ensure it will become a singleton. Also, making smaller `func` will let us unit test so such small mistakes can be caught quickly and in isolation, as there is no need to even start k8s TestEnv for that.

Reply:
What I think you are missing here is adding one line:

```
if currentDSCI.CreationTimestamp.Before(&earliestDSCI.CreationTimestamp) || currentDSCI.CreationTimestamp.Equal(&earliestDSCI.CreationTimestamp) {
                markedWrongDSCI = append(markedWrongDSCI, earliestDSCI)
				earliestDSCI = &currentDSCI
			} else {
				markedWrongDSCI = append(markedWrongDSCI, currentDSCI)
			}
```

but I would still consider making a webhook instead :)

Reply:
did bit fix for the loop logic^ 

---
the reason continued with this PR mostly because we have not finalize the webhook part.
the goal is cleanup all the checks in controller and move it to webhook, but that will take a bit more time to get the code out esp. into downstream.
since this has been reported as a bug and if a quick fix and patch the panic, it will buy us some time in case anyone forked the change, and the old plan is to have it in odh 2.5 then maybe in odh 2.6(+) we have webhook in place.
but now i am not certain when we are going to have odh 2.5